dtfs-central-api/src/services/repositories/utilisation-reports-repo/utilisation-reports-repo.ts

```diff
@@ -58,14 +72,17 @@ export const getOpenReportsBeforeReportPeriodForBankId = async (reportPeriodStar
           $or: [
             { year: { $lt: reportPeriodStart.year } },
             {
-              $and: [
-                { year: { $eq: reportPeriodStart.year } },
-                { month: { $lt: reportPeriodStart.month } }
-              ],
+              $and: [{ year: { $eq: reportPeriodStart.year } }, { month: { $lt: reportPeriodStart.month } }],
             },
           ],
         },
       ],
     })
     .toArray();
 };
+
+export const getCurrentUtilisationReportByBankIdAndReportPeriod = async (
+  bankId: string,
+  currentReportPeriod: ReportPeriod,
+): Promise<UtilisationReport | null> =>
+  await getUtilisationReportDetailsByBankIdMonthAndYear(bankId, currentReportPeriod.start.month, currentReportPeriod.start.year);
```

Comment:
This feels a bit redundant now, and having "Current" in the function name isn't guaranteed because it's dependent on a value we pass. We could instead just have this as
```
export const getUtilisationReportByBankIdAndReportPeriod = async (
  bankId: string,
  reportPeriod: ReportPeriod,
): Promise<UtilisationReport | null> =>
  await getUtilisationReportDetailsByBankIdMonthAndYear(bankId, reportPeriod.start.month, reportPeriod.start.year);
```
but then we may as well just call `getUtilisationReportDetailsByBankIdMonthAndYear`. I wouldn't mind if we wanted to keep this as a helper, but I think we should at least remove `Current` from the function and param name.

Reply:
I agree that it is redundant now, although I feel that getting reports by report period rather than month and year is the better way to go given our new way of storing reports. I would prefer to replace the 'by month and year' function with this 'by report period', but what do you think?

Reply:
I decided instead to further refine how this function works. It not uses all the information from the `reportPeriod` to select the report rather than just using the `month` and `year`